VideoLocker/src/main/java/org/edx/mobile/view/CourseDiscussionResponsesFragment.java

```diff
@@ -65,13 +63,15 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
     public void onViewCreated(View view, Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
 
+        commentsLoader = new CommentsLoader(discussionThread.getType() == DiscussionThread.ThreadType.QUESTION);
+
         // Using application context to prevent activity leak since adapter is retained across config changes
         courseDiscussionResponsesAdapter = new CourseDiscussionResponsesAdapter(
                 getActivity().getApplicationContext(), this, discussionThread);
         controller = InfiniteScrollUtils.configureRecyclerViewWithInfiniteList(discussionResponsesRecyclerView, courseDiscussionResponsesAdapter, new InfiniteScrollUtils.PageLoader<DiscussionComment>() {
             @Override
             public void loadNextPage(@NonNull InfiniteScrollUtils.PageLoadCallback<DiscussionComment> callback) {
-                getCommentList(callback);
+                commentsLoader.loadComments(callback);
```

Comment:
If `CommentsLoader` is meant for single-instance usage, then we might as well have it implement the `InfiniteScrollUtils.PageLoader<DiscussionComment>` interface instead of creating a disposable anonymous class for this that only delegates to the loader.